presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java

```diff
@@ -300,18 +300,16 @@ public void testAggregationPushdown()
         assertPushedDown("SELECT regionkey, min(nationkey) FROM nation GROUP BY regionkey");
         assertPushedDown("SELECT regionkey, max(nationkey) FROM nation GROUP BY regionkey");
         assertPushedDown("SELECT regionkey, sum(nationkey) FROM nation GROUP BY regionkey");
-        assertPushedDown(
-                "SELECT regionkey, avg(nationkey) FROM nation GROUP BY regionkey",
-                "SELECT regionkey, avg(CAST(nationkey AS double)) FROM nation GROUP BY regionkey");
+        assertPushedDown("SELECT regionkey, avg(nationkey) FROM nation GROUP BY regionkey");
 
         try (AutoCloseable ignoreTable = withTable("tpch.test_aggregation_pushdown", "(short_decimal decimal(9, 3), long_decimal decimal(30, 10))")) {
             execute("INSERT INTO tpch.test_aggregation_pushdown VALUES (100.000, 100000000.000000000)");
             execute("INSERT INTO tpch.test_aggregation_pushdown VALUES (123.321, 123456789.987654321)");
 
-            assertPushedDown("SELECT min(short_decimal), min(long_decimal) FROM test_aggregation_pushdown", "SELECT 100.000, 100000000.000000000");
-            assertPushedDown("SELECT max(short_decimal), max(long_decimal) FROM test_aggregation_pushdown", "SELECT 123.321, 123456789.987654321");
-            assertPushedDown("SELECT sum(short_decimal), sum(long_decimal) FROM test_aggregation_pushdown", "SELECT 223.321, 223456789.987654321");
-            assertPushedDown("SELECT avg(short_decimal), avg(long_decimal) FROM test_aggregation_pushdown", "SELECT 223.321 / 2, 223456789.987654321 / 2");
+            assertPushedDown("SELECT min(short_decimal), min(long_decimal) FROM test_aggregation_pushdown");
+            assertPushedDown("SELECT max(short_decimal), max(long_decimal) FROM test_aggregation_pushdown");
+            assertPushedDown("SELECT sum(short_decimal), sum(long_decimal) FROM test_aggregation_pushdown");
+            assertPushedDown("SELECT avg(short_decimal), avg(long_decimal) FROM test_aggregation_pushdown");
```

Comment:
Can you also please rename this method from `assertPushedDown` to `assertAggregationPushDown`? (in separate commit)

Reply:
I intentionally named the function `assertPushedDown`, because it was not aggregation pushdown specific.
I want it to remain as such. The fact it currently uses aggregate pushdown toggle to calculate the expected results
should be considered an implementation detail. It could as well do `OR rand() == 42` and/or use any other toggles.

Reply:
> The fact it currently uses aggregate pushdown toggle to calculate the expected results
should be considered an implementation detail.

This determines the use case, you cannot use that toggle for predicate pushdown. If you could  use `OR rand() == 42` then I would be ok to use generic `assertPushedDown` name.

Reply:
Adding `OR rand() == 42` to arbitrary queries would be hard. Unless there's a easy toggle like another session property I'd rather have a separate `assertPredicatePushedDown` which could handle that specifically.